PBR/src/PBR_Renderer.cpp

```diff
@@ -480,21 +480,45 @@ void PBR_Renderer::PrecomputeCubemaps(IDeviceContext* pCtx,
         float4x4 Rotation;
 
         float Roughness;
-        float EnvMapDim;
+        float EnvMapWidth;
+        float EnvMapHeight;
         uint  NumSamples;
-        float Dummy;
+    };
+
+    auto GetIBL_PSO = [&](IBL_FEATURE_FLAGS FeatureFlags, TEXTURE_FORMAT RTVFormat) -> RefCntAutoPtr<IPipelineState>& {
+        auto Iter = m_IBL_PSOCache.find(IBL_PSOKey{FeatureFlags, RTVFormat});
+        if (Iter != m_IBL_PSOCache.end())
+            return Iter->second;
+
+        auto Condition = m_IBL_PSOCache.emplace(IBL_PSOKey{FeatureFlags, RTVFormat}, RefCntAutoPtr<IPipelineState>{});
+        return Condition.first->second;
```

Comment:
There is a much shorter version that does exactly the same:
`m_IBL_PSOCache.emplace[IBL_PSOKey{FeatureFlags, RTVFormat}]`